src/components/app-configure/auth/PasswordlessAuth.vue

```diff
@@ -37,45 +80,75 @@ function handleUpdate(type: 'javascriptOrigin' | 'redirectUri', value: string) {
           Learn More...
         </a>
       </template>
-      <VStack
-        direction="row"
-        align="center"
-        sm-direction="column"
-        sm-align="start"
-        gap="1rem"
-      >
-        <label for="passwordless-javascript-origin">JavaScript Origin</label>
-        <VTextField
-          id="passwordless-javascript-origin"
-          :model-value="passwordless.javascriptOrigin"
-          class="passwordless-input"
-          no-message
-          @update:model-value="handleUpdate('javascriptOrigin', $event)"
-        />
-      </VStack>
-      <VStack
-        direction="row"
-        align="center"
-        sm-direction="column"
-        sm-align="start"
-        gap="1rem"
-      >
-        <label for="passwordless-redirect-uri">Redirect URI</label>
-        <VTextField
-          id="passwordless-redirect-uri"
-          :model-value="passwordless.redirectUri"
-          class="passwordless-input"
-          no-message
-          @update:model-value="handleUpdate('redirectUri', $event)"
-        />
-      </VStack>
+      <form @submit.prevent="handleSave">
+        <VStack direction="column" gap="1rem" class="flex-grow">
+          <VStack
+            direction="row"
+            align="center"
+            sm-direction="column"
+            sm-align="start"
+            gap="1rem"
+          >
+            <label for="passwordless-javascript-origin"
+              >JavaScript Origin</label
+            >
+            <VTextField
+              id="passwordless-javascript-origin"
+              v-model.trim="passwordless.javascriptOrigin"
+              class="passwordless-input"
+              :message-type="
+                isEdited &&
+                passwordless.redirectUri &&
+                !isValidJavascriptOrigin()
+                  ? 'error'
+                  : ''
+              "
+              message="Invalid javascript origin - must be a valid url"
+              @blur="isEdited = true"
+            />
+          </VStack>
+          <VStack
+            direction="row"
+            align="center"
+            sm-direction="column"
+            sm-align="start"
+            gap="1rem"
+          >
+            <label for="passwordless-redirect-uri">Redirect URI</label>
+            <VTextField
+              id="passwordless-redirect-uri"
+              v-model.trim="passwordless.redirectUri"
+              class="passwordless-input"
+              :message-type="
+                isEdited &&
+                passwordless.javascriptOrigin &&
+                !isValidRedirectUri()
+                  ? 'error'
+                  : ''
```

Comment:
Move expression to a computed value.

Reply:
Resolved at https://github.com/arcana-network/developer-dashboard/pull/128/commits/a6db5581e3f2a70add936a12b0cc2e14f879b209